connectors/src/connectors/google_drive/lib/google_drive_api.ts

```diff
@@ -39,3 +40,31 @@ export async function getGoogleDriveObject(
     throw e;
   }
 }
+
+const cachedGetGoogleDriveObject = cacheWithRedis<
+  GoogleDriveObjectType | null,
+  [OAuth2Client, string, number, string | number]
+>(
+  _getGoogleDriveObject,
+  (_, driveObjectId, connectorId, memoizationKey) => {
+    return `${connectorId}:${driveObjectId}:${memoizationKey}`;
+  },
+  60 * 10 * 1000
+);
+
+export async function getGoogleDriveObject(
+  authCredentials: OAuth2Client,
+  driveObjectId: string,
+  connectorId?: number,
+  memoizationKey?: string | number
```

Comment:
nit: would put that in a cache object => clearer readability, if cache / if not cache, + obvious that memoization&connectorId are here for the sole purpose of caching
```suggestion
   cache?: { connectorId: number,
  memoizationKey: string | number }
```

Reply:
definitely 💯